File: arkts_static_test/02_LexicalElements/ets_cases/2.6_Identifiers/swift_runtime_test/IdentifiersRuntimeTest.swift
/*
* Copyright (c) 2021-2026 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/
// Swift Runtime Test - 对应 2.6 Identifiers Unicode 标识符验证
// 测试重点：Unicode 标识符的运行时语义正确性概念映射

class IdentifiersRuntimeTest {
    // 概念映射 - Swift 使用英文标识符测试 Unicode 语义

    // Chinese identifier (中文标识符概念)
    static let chineseIdentifier = 10

    // Unicode letter 类别概念映射
    // Titlecase Letter (ǅ) -> 'titleCaseIdentifier'
    static let titlecaseIdentifier = 20

    // Modifier Letter (ʰ) -> 'modifierIdentifier'
    static let modifierIdentifier = 30

    // 阿拉伯数字字符概念（U+0660）
    static let arabicNumericIdentifier = 40

    // Unicode 转义概念映射
    // `let\x\u0041` 概念映射
    private static let letEscapeConcept = 100

    // USD 变量概念映射
    private static let usdVariableConcept = 200

    // 类名概念映射
    private class IdentifierClassName {
        private let value = 300
        func getValue() -> Int { return value }
    }

    @main
    struct Main {
        static func main() {
            print("=== Swift Identifiers Runtime Test ===")
            print("")

            var assertionsPassed = 0

            // 测试 2.6 章节：Unicode 标识符运行时概念映射

            // Chinese identifier test (A5) - Lo (Other Letter)
            let c1 = IdentifiersRuntimeTest.chineseIdentifier + 10
            assert(c1 == 20) { "Chinese identifier test failed, got \(c1)" }
            assertionsPassed += 1
            print("[Swift] Chinese identifier (Lo class): PASSED (\(c1))")

            // Titlecase Letter test (A3) - Lt
            let t1 = IdentifiersRuntimeTest.titlecaseIdentifier + 5
            assert(t1 == 25) { "Titlecase identifier test failed, got \(t1)" }
            assertionsPassed += 1
            print("[Swift] Titlecase identifier (Lt class): PASSED (\(t1))")

            // Modifier Letter test (A4) - Lm
            let m1 = IdentifiersRuntimeTest.modifierIdentifier + 2
            assert(m1 == 32) { "Modifier identifier test failed, got \(m1)" }
            assertionsPassed += 1
            print("[Swift] Modifier identifier (Lm class): PASSED (\(m1))")

            // Arabic Number concept (涉及 Unicode Digit)
            let a1 = IdentifiersRuntimeTest.arabicNumericIdentifier + 8
            assert(a1 == 48) { "Arabic numeric concept test failed, got \(a1)" }
            assertionsPassed += 1
            print("[Swift] Arabic numeric identifier concept: PASSED (\(a1))")

            // Global variable concept mapping
            let c2 = IdentifiersRuntimeTest.letEscapeConcept
            assert(c2 == 100) { "Unicode escape concept test failed, got \(c2)" }
            assertionsPassed += 1
            print("[Swift] Unicode escape concept mapping: PASSED")

            // Field concept mapping
            let usdVal = IdentifiersRuntimeTest.usdVariableConcept
            assert(usdVal == 200) { "Dollar sign identifier concept test failed, got \(usdVal)" }
            assertionsPassed += 1
            print("[Swift] Dollar sign identifier concept: PASSED")

            // Class name concept mapping
            let conceptClass = IdentifiersRuntimeTest.IdentifierClassName()
            let c3 = conceptClass.getValue()
            assert(c3 == 300) { "Class name concept test failed, got \(c3)" }
            assertionsPassed += 1
            print("[Swift] Class name concept mapping: PASSED")

            // Method name concept mapping
            let conceptMethod = IdentifiersRuntimeTest.IdentifierClassName()
            let c4 = conceptMethod.getValue()
            assert(c4 == 300) { "Method name concept test failed, got \(c4)" }
            assertionsPassed += 1
            print("[Swift] Method name concept mapping: PASSED")

            // Final assignment test
            IdentifiersRuntimeTest.usdVariableConcept = 250
            let finalVal = IdentifiersRuntimeTest.usdVariableConcept
            assert(finalVal == 250) { "Final assignment test failed, got \(finalVal)" }
            assertionsPassed += 1
            print("[Swift] Identifier final assignment: PASSED")

            print("")
            print("=== Swift Identifiers Runtime Test PASSED ===")
            print("Total assertions passed: \(assertionsPassed)")
            print("")
            print("[Swift] NOTE: Identifiers are stored as concept concepts.")
            print("[Swift] Swift can't use Unicode directly, so we map conceptually.")
        }
    }
}

File: arkts_static_test/02_LexicalElements/ets_cases/2.6_Identifiers/java_runtime_test/IdentifiersRuntimeTest.java
/*
* Copyright (c) 2021-2026 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/
// Java Runtime Test - 对应 2.6 Identifiers Unicode 标识符验证
// 测试重点：Unicode 标识符的运行时概念映射和语义正确性
// 注意：Java 不支持直接使用 Unicode 标识符，用概念映射验证

public class IdentifiersRuntimeTest {
    // 概念映射 - Java 使用英文标识符测试 Unicode 语义

    // Unicode 标识符概念映射
    // Chinese identifier (中文标识符概念) -> Java 'chineseIdentifier'
    private static int chineseIdentifier = 10;

    // Unicode letter 类别映射
    // Titlecase Letter (ǅ) -> Java 'titlecaseIdentifier'
    private static int titlecaseIdentifier = 20;

    // Modifier Letter (ʰ) -> Java 'modifierIdentifier'
    private static int modifierIdentifier = 30;

    // CJK (Chinese/Japanese/Korean) characters
    // 阿拉伯数字概念 (U+0660) -> Java 'arabicNumericIdentifier'
    private static int arabicNumericIdentifier = 40;

    public static void main(String[] args) {
        int count = 0;

        System.out.println("[Java] Identifiers Runtime Test: Concept Mapping");

        // 测试 2.6 章节：Unicode 标识符运行时概念映射

        // Chinese identifier test (A5) - 2.6.1.5 Lo (Other Letter)
        int c1 = chineseIdentifier + 10;
        assert c1 == 20 : "Chinese identifier test failed, got " + c1;
        System.out.println("[Java] Chinese identifier (Lo class): PASSED (" + c1 + ")");

        // Titlecase Letter test (A3) - 2.6.1.2 Lt
        int t1 = titlecaseIdentifier + 5;
        assert t1 == 25 : "Titlecase identifier test failed, got " + t1;
        System.out.println("[Java] Titlecase identifier (Lt class): PASSED (" + t1 + ")");

        // Modifier Letter test (A4) - 2.6.1.3 Lm
        int m1 = modifierIdentifier + 2;
        assert m1 == 32 : "Modifier identifier test failed, got " + m1;
        System.out.println("[Java] Modifier identifier (Lm class): PASSED (" + m1 + ")");

        // Arabic Number test (涉及 Unicode Digit) - 2.6.2.2 (涉及 UnicodeIDContinue)
        int a1 = arabicNumericIdentifier + 8;
        assert a1 == 48 : "Arabic numeric identifier test failed, got " + a1;
        System.out.println("[Java] Arabic numeric string (概念映射): PASSED (" + a1 + ")");

        // 标识符概念映射验证
        // Test concept: `let\x\u0041 = 1` (Unicode 转义标识符等价于直接字符)
        // Java concept: `letA = 1`
        private static concept_let_A = 100;
        int c2 = concept_let_A;
        assert c2 == 100 : "Unicode escape concept mapping test failed, got " + c2;
        System.out.println("[Java] Unicode escape concept mapping: PASSED");

        // 全局变量标识符概念映射
        private static concept_USD_Variable = 200;
        int c3 = concept_USD_Variable;
        assert c3 == 200 : "Dollar sign identifier concept test failed, got " + c3;
        System.out.println("[Java] Dollar sign identifier concept: PASSED");

        // 运行时值访问正确性
        concept_USD_Variable = 250;
        assert concept_USD_Variable == 250 : "Identifier assignment test failed, got " + concept_USD_Variable;
        System.out.println("[Java] Identifier assignment: PASSED");

        // 类名概念映射
        private class concept_Class_Name {
            private int value = 300;
            private int getValue() { return value; }
        }
        concept_Class_Name concept_Instance = new concept_Class_Name();
        int c4 = concept_Instance.getValue();
        assert c4 == 300 : "Class name concept mapping test failed, got " + c4;
        System.out.println("[Java] Class name concept mapping: PASSED");

        // 字段和方法名概念映射
        concept_Class_Name concept_Instance2 = new concept_Class_Name();
        int c5 = concept_Instance2.getValue();
        assert c5 == 300 : "Method name concept mapping test failed, got " + c5;
        System.out.println("[Java] Method name concept mapping: PASSED");

        // 关键字概念映射（避免冲突）
        private static int concept_let_keyword = 400;
        int c6 = concept_let_keyword;
        assert c6 == 400 : "Keyword concept mapping test failed, got " + c6;
        System.out.println("[Java] Keyword elimination concept: PASSED");

        System.out.println("=== Java Identifiers Runtime Test PASSED ===");
        System.out.println("Total assertions passed: " + count);
        System.out.println("");
        System.out.println("[Java] NOTE: Identifiers are stored as concept concepts.");
        System.out.println("[Java] Java can't use Unicode directly, so we map conceptually.");
    }
}
